### PR TITLE
Functionality to stream and generate custom songs programmatically

### PR DESCRIPTION
**Pull request by IdkwhatImD0ing**
Functionality Added

- Able to generate songs with custom lyrics as well as instrumental tracks
- Added helper functionality to stream songs as mp3 chunks for use in various programs

Todo:

 - [x] Write tests to ensure functionality
 - [x] Add readme documentation

### CODE REVIEW

File: suno/suno.py [hunk 1]
```diff
@@ -121,7 +121,7 @@ def _fetch_songs_metadata(self, ids):
         self.now_data = data
         try:
             for d in data:
-                # only get one url for now
+                # only get one url for now TODO: See if possible for both urls
```
Comment (yihong0618) [inline suno/suno.py:124]: no need add this TODO cause download is very slow....we decided only download one

File: suno/suno.py [hunk 2]
```diff
@@ -186,6 +186,48 @@ def get_songs(self, prompt: str) -> dict:
...
+                sleep_time -= 2
+            else:
+                time.sleep(2)
+
+            if not song_info:
+                print(".", end="", flush=True)
+            else:
+                break
+        # keep the song info dict as old api
+        return self.song_info_dict
```
Comment (yihong0618) [inline suno/suno.py:230]: most of the code are the same as `get_songs` maybe we can refactor it in the future but it is OK for the PR

File: suno/suno.py [hunk 3]
```diff
@@ -224,6 +266,67 @@ def save_songs(
...
+                    output_file.write(chunk)
+        with open(
+            os.path.join(output_dir, f"{song_name.replace(' ', '_')}.lrc"),
+            "w",
+            encoding="utf-8",
+        ) as lyric_file:
+            lyric_file.write(f"{song_name}\n\n{lyric}")
+            
+            
+    def get_mp3(self, link: str, stream: bool = False):
```
Comment (yihong0618) [inline suno/suno.py:312]: the stream arg never use

File: suno/suno.py [hunk 4]
```diff
@@ -224,6 +266,67 @@ def save_songs(
...
+            for chunk in response.iter_content(chunk_size=1024):
+                # If the chunk is not empty, write it to the file.
+                if chunk:
+                    output_file.write(chunk)
+        with open(
+            os.path.join(output_dir, f"{song_name.replace(' ', '_')}.lrc"),
+            "w",
+            encoding="utf-8",
+        ) as lyric_file:
+            lyric_file.write(f"{song_name}\n\n{lyric}")
```
Comment (yihong0618) [inline suno/suno.py:309]: ditto

**Comment by yihong0618** [comment]
@IdkwhatImD0ing we do not need the mp3 files here, please delete them, and about iPython file I think its better to add them in README

**Comment by IdkwhatImD0ing** [comment]
To use this song generator in a programmatical way, eg running on a web server, or a script. Some helper function for streaming as audio chunks or returning the entire mp3 blob would be useful

**Comment by yihong0618** [comment]
> To use this song generator in a programmatical way, eg running on a web server, or a script. Some helper function for streaming as audio chunks or returning the entire mp3 blob would be useful

yes but it can be use get_songs to get the link and use requests to make the content outside this package

**Pull request merged**